asciidoctorj-core/src/main/java/org/asciidoctor/extension/Processor.java

```diff
@@ -16,25 +17,29 @@
 
 public class Processor {
 
-	protected RubyHash config;
-    protected Ruby rubyRuntime;
+	protected Map<String, Object> config;
 
     public Processor(Map<String, Object> config) {
-        this.rubyRuntime = JRubyRuntimeContext.get();
-        this.config = RubyHashUtil.convertMapToRubyHashWithSymbols(rubyRuntime, config);
+        this.config = config;
     }
 
-    public void update_config(Map<Object, Object> config) {
+    public void update_config(Map<String, Object> config) {
     	this.config.putAll(config);
     }
     
-    public Map<Object, Object> getConfig() {
+    public Map<String, Object> getConfig() {
     	return this.config;
     }
 
-    public RubyHash setConfig(Map<String, Object> config) {
-        this.config = RubyHashUtil.convertMapToRubyHashWithSymbols(rubyRuntime, config);
-        return null;
+    public void setConfig(Map<String, Object> config) {
+        if (config instanceof RubyHashMapDecorator) {
+            // ProcessorProxy has already taken over, replace config
+            this.config.clear();
+            this.config.putAll(config);
+        } else {
+            // Processor is still independent, simply set the member
+            this.config = config;
+        }
```

Comment:
I think that a concrete implementation might want to try to invoke this method in the constructor.
(We should probably add a test case for that, YellStaticBlock currently does it in a way that not every other implementor might go to forward the config in the super constructor invocation: https://github.com/asciidoctor/asciidoctorj/blob/master/asciidoctorj-core/src/test/java/org/asciidoctor/extension/YellStaticBlock.java#L18 )

What do you think about keeping `setConfig(Map)` public but final and add a new package private method `setConfigFinalized()` that is called by the proxies and sets a flag to indicate that the config must not be reset anymore?
`setConfig(Map)` would throw an `IllegalStateException` if called after `setConfigFinalized()`.

(Nevertheless it still feels a little hacky and I'd prefer a way to already make the compiler refuse such a situation, but I don't see how to do it.)